File: 03_Spring/Backend_Course/assignment1/src/main/java/com/example/assignment1/board/impl/BoardServiceImpl.java
package com.example.assignment1.board.impl;

import com.example.assignment1.board.BoardService;
import com.example.assignment1.board.BoardVO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BoardServiceImpl implements BoardService {
    private BoardDAO boardDAO;

    @Autowired
    public BoardServiceImpl(BoardDAO boardDAO) {
        this.boardDAO = boardDAO;
    }

    @Override
    public void insertBoard(BoardVO vo) {
        boardDAO.insertBoard(vo);
    }

    @Override
    public void deleteBoard(BoardVO vo) {
        boardDAO.deleteBoard(vo);
    }

    @Override
    public void updateBoard(BoardVO vo) {
        boardDAO.updateBoard(vo);
    }

    @Override
    public BoardVO getBoard(BoardVO vo) {
        return boardDAO.getBoard(vo);
    }

    @Override
    public List<BoardVO> getBoardContentList(BoardVO vo) { return boardDAO.getBoardContentList(vo); }
    @Override
    public List<BoardVO> getBoardList(BoardVO vo) {
        return boardDAO.getBoardList(vo);
    }
}

File: 03_Spring/Backend_Course/assignment1/src/main/java/com/example/assignment1/board/impl/BoardServiceImpl.kt
package com.example.assignment1.board.impl

import com.example.assignment1.board.BoardService
import com.example.assignment1.board.BoardVO
import org.springframework.stereotype.Service

@Service
open class BoardServiceImpl(private val boardDAO: BoardDAO): BoardService {
    override fun insertBoard(vo: BoardVO) = boardDAO.insertBoard(vo)
    override fun deleteBoard(vo: BoardVO) = boardDAO.deleteBoard(vo)
    override fun updateBoard(vo: BoardVO) = boardDAO.updateBoard(vo)
    override fun getBoard(vo: BoardVO): BoardVO? = boardDAO.getBoard(vo)
    override fun getBoardContentList(vo: BoardVO): List<BoardVO>? = boardDAO.getBoardContentList(vo)
    override fun getBoardList(vo: BoardVO?): List<BoardVO>? = boardDAO.getBoardList(vo)
}